Aggregate small messages while streaming (#14)
awakesecurity/grpc-mqtt: README.md

The RemoteClient performs the actual gRPC requests on behalf of the Client. Similarily to the Client, the RemoteClient code can be generated using `mqttRemoteClientMethodMap`. See Test/ProtoRemoteClients.hs for an example. The resulting `MethodMap` is a mapping from gRPC method names to a function for making that request. These maps can be combined if you have multiple gRPC servers running on the machine.

General usage:
```haskell
withGRPCClient myGRPCClientConfig $ \grpcClient -> do
  methodMap <- addHelloRemoteClientMethodMap grpcClient
  runRemoteClient logger myMQTTConfig baseTopic methodMap
```
Using multiple servers:
```haskell
methodMapAH <- addHelloRemoteClientMethodMap grpcClient1
methodMapMG <- multGoodbyeRemoteClientMethodMap grpcClient2
let methodMap = methodMapAH <> methodMapMG
runRemoteClient logger myMQTTConfig baseTopic methodMap
```

**Batching**

Typically, each message transmitted through GRPC method calls result in one or more MQTT packets published over MQTT. A
packet size limit is configured in `MQTTGRPCConfig.mqttMsgSizeLimit`. If a message is larger than this limit, it will be

> split to multiple packets and then those packets are published.

Reviewer: ```suggestion
split into multiple packets and then those packets are published.
```

> Streaming RPCs that transmit many small messages in a short time can get a performance improvement by enabling

Reviewer: ```suggestion
The performance of streaming RPCs that transmit many small messages in a short time window can be improved, dramatically, by enabling
```

> batching. When batching is enabled, the sender accumulates many messages to one packet and then flushes them in a single

Reviewer: ```suggestion
batching. When batching is enabled, the sender accumulates many messages into one packet and then flushes them in a single
```
